build_tools/checks.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+set -e
+
+repo_root="$(dirname "$0")/.."
```

Comment:
should also run clippy I think. I guess that one is not enforced in the CI today,
but same goes for formatting and warnings in Rustdoc.
I think we're not far from enforcing all checks.